Zstd fails to install

Hi. I try to install via pip and get the following error:
Visual C++ also installed. 
Setuptools version: 40.5.0
How should I resolve this issue?
```
C:\Users\Emil>pip3 install zstd
Collecting zstd
  Using cached https://files.pythonhosted.org/packages/08/ea/693d977411af8f9a43aee43a092e8b30542cdb9c5a326739ff844549cf93/zstd-1.3.5.1.tar.gz
Installing collected packages: zstd
  Running setup.py install for zstd ... error
    Complete output from command c:\users\emil\appdata\local\programs\python\python36\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\Emil\\AppData\\Local\\Temp\\pip-install-drb0myv4\\zstd\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\Emil\AppData\Local\Temp\pip-record-vef6jxhh\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_ext
    building 'zstd' extension
    creating build
    creating build\temp.win-amd64-3.6
    creating build\temp.win-amd64-3.6\Release
    creating build\temp.win-amd64-3.6\Release\zstd
    creating build\temp.win-amd64-3.6\Release\zstd\lib
    creating build\temp.win-amd64-3.6\Release\zstd\lib\compress
    creating build\temp.win-amd64-3.6\Release\zstd\lib\decompress
    creating build\temp.win-amd64-3.6\Release\zstd\lib\common
    creating build\temp.win-amd64-3.6\Release\src
    C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.15.26726\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Ic:\users\emil\appdata\local\programs\python\python36\include -Ic:\users\emil\appdata\local\programs\python\python36\include "-IC:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.15.26726\include" /Tczstd/lib/compress/zstd_compress.c /Fobuild\temp.win-amd64-3.6\Release\zstd/lib/compress/zstd_compress.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    zstd_compress.c
    zstd/lib/compress/zstd_compress.c(14): fatal error C1083: Datei (Include) kann nicht ge”ffnet werden: "string.h": No such file or directory
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\BuildTools\\VC\\Tools\\MSVC\\14.15.26726\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2

    ----------------------------------------
Command "c:\users\emil\appdata\local\programs\python\python36\python.exe -u -c "import setuptools, tokenize;__file__='C:\\Users\\Emil\\AppData\\Local\\Temp\\pip-install-drb0myv4\\zstd\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record C:\Users\Emil\AppData\Local\Temp\pip-record-vef6jxhh\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in C:\Users\Emil\AppData\Local\Temp\pip-install-drb0myv4\zstd\
```

Not trying to necro this, but having similar issues and despite the reinstall of both vs 9 for py2 and 17, I got  no where beyond an incredibly verbose error message. Figured I'd pop by here and see how I could go about fixing this?

```
C:\WINDOWS\system32>c:\Python27\python.exe -m pip install zstd
Collecting zstd
  Using cached https://files.pythonhosted.org/packages/08/ea/693d977411af8f9a43aee43a092e8b30542cdb9c5a326739ff844549cf93/zstd-1.3.5.1.tar.gz
Installing collected packages: zstd
  Running setup.py install for zstd ... error
    Complete output from command c:\Python27\python.exe -u -c "import setuptools, tokenize;__file__='c:\\users\\mac\\appdata\\local\\temp\\pip-install-khylzr\\zstd\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record c:\users\mac\appdata\local\temp\pip-record-kijliv\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_ext
    building 'zstd' extension
    creating build
    creating build\temp.win-amd64-2.7
    creating build\temp.win-amd64-2.7\Release
    creating build\temp.win-amd64-2.7\Release\zstd
    creating build\temp.win-amd64-2.7\Release\zstd\lib
    creating build\temp.win-amd64-2.7\Release\zstd\lib\compress
    creating build\temp.win-amd64-2.7\Release\zstd\lib\decompress
    creating build\temp.win-amd64-2.7\Release\zstd\lib\common
    creating build\temp.win-amd64-2.7\Release\src
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/compress/zstd_compress.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/compress/zstd_compress.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    zstd_compress.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/compress/zstdmt_compress.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/compress/zstdmt_compress.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    zstdmt_compress.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/compress/zstd_fast.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/compress/zstd_fast.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    zstd_fast.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/compress/zstd_double_fast.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/compress/zstd_double_fast.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    zstd_double_fast.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/compress/zstd_lazy.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/compress/zstd_lazy.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    zstd_lazy.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/compress/zstd_opt.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/compress/zstd_opt.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    zstd_opt.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/compress/zstd_ldm.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/compress/zstd_ldm.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    zstd_ldm.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/compress/fse_compress.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/compress/fse_compress.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    fse_compress.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/compress/huf_compress.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/compress/huf_compress.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    huf_compress.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/compress/hist.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/compress/hist.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    hist.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/decompress/zstd_decompress.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/decompress/zstd_decompress.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    zstd_decompress.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/common/fse_decompress.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/common/fse_decompress.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    fse_decompress.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/decompress/huf_decompress.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/decompress/huf_decompress.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    huf_decompress.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/common/entropy_common.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/common/entropy_common.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    entropy_common.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/common/zstd_common.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/common/zstd_common.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    zstd_common.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/common/xxhash.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/common/xxhash.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    xxhash.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/common/error_private.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/common/error_private.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    error_private.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tczstd/lib/common/pool.c /Fobuild\temp.win-amd64-2.7\Release\zstd/lib/common/pool.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    pool.c
    C:\Users\Mac\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\Python27\include -Ic:\Python27\PC /Tcsrc/python-zstd.c /Fobuild\temp.win-amd64-2.7\Release\src/python-zstd.obj /Ox /DVERSION="\"1.3.5.1\"" /Izstd\lib /Izstd\lib\common /Izstd\lib\compress /Izstd\lib\decompress
    python-zstd.c
    src/python-zstd.c(75) : warning C4267: '=' : conversion from 'size_t' to 'uint32_t', possible loss of data
    src/python-zstd.c(109) : error C2065: 'uint64_t' : undeclared identifier
    src/python-zstd.c(109) : error C2146: syntax error : missing ';' before identifier 'dest_size'
    src/python-zstd.c(109) : error C2065: 'dest_size' : undeclared identifier
    src/python-zstd.c(110) : error C2143: syntax error : missing ';' before 'type'
    src/python-zstd.c(111) : error C2275: 'size_t' : illegal use of this type as an expression
            src/python-zstd.c : see declaration of 'size_t'
    src/python-zstd.c(111) : error C2146: syntax error : missing ';' before identifier 'cSize'
    src/python-zstd.c(111) : error C2065: 'cSize' : undeclared identifier
    src/python-zstd.c(121) : error C2065: 'dest_size' : undeclared identifier
    src/python-zstd.c(121) : error C2065: 'uint64_t' : undeclared identifier
    src/python-zstd.c(121) : error C2146: syntax error : missing ';' before identifier 'ZSTD_getDecompressedSize'
    src/python-zstd.c(122) : error C2065: 'dest_size' : undeclared identifier
    src/python-zstd.c(126) : error C2065: 'dest_size' : undeclared identifier
    src/python-zstd.c(132) : error C2065: 'cSize' : undeclared identifier
    src/python-zstd.c(132) : error C2065: 'dest_size' : undeclared identifier
    src/python-zstd.c(132) : warning C4267: '=' : conversion from 'size_t' to 'int', possible loss of data
    src/python-zstd.c(135) : error C2065: 'cSize' : undeclared identifier
    src/python-zstd.c(136) : error C2065: 'cSize' : undeclared identifier
    src/python-zstd.c(137) : error C2065: 'error' : undeclared identifier
    src/python-zstd.c(138) : error C2065: 'cSize' : undeclared identifier
    src/python-zstd.c(138) : error C2065: 'dest_size' : undeclared identifier
    src/python-zstd.c(139) : error C2065: 'uint64_t' : undeclared identifier
    src/python-zstd.c(139) : error C2146: syntax error : missing ')' before identifier 'cSize'
    src/python-zstd.c(139) : error C2059: syntax error : ')'
    src/python-zstd.c(140) : error C2065: 'error' : undeclared identifier
    src/python-zstd.c(144) : error C2065: 'error' : undeclared identifier
    src/python-zstd.c(324) : warning C4113: 'PyObject *(__cdecl *)(PyObject *)' differs in parameter lists from 'PyCFunction'
    src/python-zstd.c(325) : warning C4113: 'PyObject *(__cdecl *)(PyObject *)' differs in parameter lists from 'PyCFunction'
    src/python-zstd.c(326) : warning C4113: 'PyObject *(__cdecl *)(PyObject *)' differs in parameter lists from 'PyCFunction'
    error: command 'C:\\Users\\Mac\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\amd64\\cl.exe' failed with exit status 2

    ----------------------------------------
Command "c:\Python27\python.exe -u -c "import setuptools, tokenize;__file__='c:\\users\\mac\\appdata\\local\\temp\\pip-install-khylzr\\zstd\\setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record c:\users\mac\appdata\local\temp\pip-record-kijliv\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in c:\users\mac\appdata\local\temp\pip-install-khylzr\zstd\
```

--- a/src/python-zstd.c
+++ b/src/python-zstd.c
@@ -28,6 +28,7 @@
  */
 
 #include <stdlib.h>
+#include <stddef.h>
 #include <Python.h>
 #include "bytesobject.h"
 #include "zstd.h"

--- a/src/python-zstd.h
+++ b/src/python-zstd.h
@@ -113,12 +113,14 @@ typedef uint8_t  BYTE;
 typedef uint16_t U16;
 typedef uint32_t U32;
 typedef  int32_t S32;
+typedef  int64_t S64;
 typedef uint64_t U64;
 #else
 typedef unsigned char       BYTE;
 typedef unsigned short      U16;
 typedef unsigned int        U32;
 typedef   signed int        S32;
+typedef   signed long long  S64;
 typedef unsigned long long  U64;
 #endif
 
@@ -127,12 +129,14 @@ typedef unsigned long long  U64;
 # if _MSC_VER >= 1600
 #  include <stdint.h>
 # else /* _MSC_VER >= 1600 */
-   typedef signed char       int8_t;
-   typedef signed short      int16_t;
-   typedef signed int        int32_t;
-   typedef unsigned char     uint8_t;
-   typedef unsigned short    uint16_t;
-   typedef unsigned int      uint32_t;
+   typedef signed char        int8_t;
+   typedef signed short       int16_t;
+   typedef signed int         int32_t;
+   typedef signed long long   int64_t;
+   typedef unsigned char      uint8_t;
+   typedef unsigned short     uint16_t;
+   typedef unsigned int       uint32_t;
+   typedef unsigned long long uint64_t;
 # endif /* _MSC_VER >= 1600 */
 #endif
 

